src/cloud/benchflow/sender/sender.go

```diff
@@ -153,51 +180,74 @@ func consumeFromTopic(t TransformerSetting) {
 						panic(err)
 						}
 					fmt.Println("Script "+s.Script+" processed")
-					launchAnalyserScript(msg.Trial_id, msg.Total_trials_num, t.Topic)
+					totalTrialsNum, _ := strconv.Atoi(msg.Total_trials_num)
+					launchAnalyserScripts(msg.Trial_id, totalTrialsNum, t.Topic)
 					}
+				consumer.CommitUpto(m)
 			}
-			waitGroup.Done()
+		consumer.Close()
+		waitGroup.Done()
```

Comment:
It's not inside the loop, it's outside. When the loop ends after being signalled to terminate, it closes the consumer.